### DISCUSSION TITLE
请教下，要实现字幕跑马灯的效果，速度固定，后期可以修改文本如何实现呢

### DISCUSSION THREAD

**Discussion by Jiiiin** [Tencent/libpag#1877] [Q&A] [answered]
pag效果如图所示，文字会逐步从右往左移动，渲染效果也是ok的，问题是动画时长是固定的，但是文字长度不定，希望文字变长后，动画也变长，这个如何实现呢。考虑过时间伸缩功能，似乎没成功。
![image](https://github.com/Tencent/libpag/assets/22162992/9bc57672-fe9b-4ea7-8df0-7bc7518a6c23)

**Comment by kevingpqi123** [comment, marked as answer]
可以通过 PAG 的组合模式来实现，创建一个空的 PAGComposition，宽度是屏幕宽度，高度是字幕需要显示的高度，比如限定同时显示多少行
（1）如果只需要播放一次，设计一个时长较长的文本跑马灯效果的 pag 文件，根据用户的输入添加 pag 文件编辑文本就可以
（2）如果需要循环播放效果，那就设计多套 pag 文件，比如 10、20、30、40等多个字符效果的 pag 文件，实际使用的过程中根据用户的文本数量匹配相应的 pag 文件即可

**Reply by Jiiiin** [reply]
感谢大佬，目前是只要播放一次，那我只能搞个时长较长的文件了。。
